aws-auth-cognito/src/test/java/featureTest/utilities/CognitoMockFactory.kt

```diff
@@ -0,0 +1,97 @@
+import aws.sdk.kotlin.services.cognitoidentityprovider.model.ForgotPasswordResponse
+import aws.sdk.kotlin.services.cognitoidentityprovider.model.SignUpRequest
+import aws.sdk.kotlin.services.cognitoidentityprovider.model.SignUpResponse
+import com.amplifyframework.auth.cognito.helpers.AuthHelper
+import com.amplifyframework.testutils.featuretest.MockResponse
+import com.amplifyframework.testutils.featuretest.ResponseType
+import com.amplifyframework.testutils.featuretest.auth.definitions.toCognitoException
+import io.mockk.CapturingSlot
+import io.mockk.coEvery
+import io.mockk.mockkObject
+import io.mockk.slot
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+
+/**
+ * Factory to mock aws sdk's cognito API calls and responses.
+ */
+class CognitoMockFactory(private val mockCognitoIPClient: CognitoIdentityProviderClient) {
+    private val captures: MutableMap<String, CapturingSlot<*>> = mutableMapOf()
+
+    fun mock(mockResponse: MockResponse) {
+        val responseObject = mockResponse.response as JsonObject
+        fun expectedException(responseObject: JsonObject) =
+            CognitoIdentityProviderException((responseObject["message"] as JsonPrimitive).content)
+
+        return when (mockResponse.apiName) {
+            "forgotPassword" -> {
+                val requestBuilderCaptor = slot<ForgotPasswordRequest>()
+
+                coEvery { mockCognitoIPClient.forgotPassword(capture(requestBuilderCaptor)) } coAnswers {
+                    if (mockResponse.responseType == ResponseType.Failure) {
+                        throw responseObject.toCognitoException()
+                    }
+                    ForgotPasswordResponse.invoke {
+                        this.codeDeliveryDetails = parseCodeDeliveryDetails(responseObject)
+                    }
+                }
+
+                captures[mockResponse.apiName] = requestBuilderCaptor
+            }
+            "signUp" -> {
+                mockkObject(AuthHelper)
+                coEvery { AuthHelper.getSecretHash(any(), any(), any()) } returns "a hash"
+
+                val requestCaptor = slot<SignUpRequest>()
+
+                coEvery { mockCognitoIPClient.signUp(capture(requestCaptor)) } coAnswers {
+                    if (mockResponse.responseType == ResponseType.Failure) {
+                        throw responseObject.toCognitoException()
+                    }
+                    SignUpResponse.invoke {
+                        this.codeDeliveryDetails = parseCodeDeliveryDetails(responseObject)
+                    }
+                }
+                captures[mockResponse.apiName] = requestCaptor
+            }
+
+            else -> throw Error("mock for ${mockResponse.apiName} not defined!")
+        }
+    }
+
+    private fun parseCodeDeliveryDetails(response: JsonObject): CodeDeliveryDetailsType {
+        val codeDeliveryDetails = response["codeDeliveryDetails"] as JsonObject
```

Comment:
Can we keep the responses similar to what is there in the [sdk models](https://github.com/awslabs/aws-sdk-swift/blob/main/codegen/sdk-codegen/aws-models/cognito-identity-provider.2016-04-18.json#L10542)? 

which is starting with the capital case.. i.e. `codeDeliveryDetails` to `CodeDeliveryDetails `

Reply:
Will this be done on this PR or a follow up PR ?

Reply:
Follow up PR, currently looking into smithy models.